source/gui/addonGui.py

```diff
@@ -303,16 +303,18 @@ def getAddonStatus(self, addon):
 			# Translators: The status shown for a newly installed addon before NVDA is restarted.
 			statusList.append(_("Install"))
 		# in some cases an addon can be expected to be disabled after install, so we want "install" to take precedence here
-		elif globalVars.appArgs.disableAddons or addon.isDisabled:
+		elif addon.isDisabled or globalVars.appArgs.disableAddons:
 			# Translators: The status shown for an addon when its currently suspended do to addons being disabled.
 			statusList.append(_("Disabled"))
 		if addon.isPendingRemove:
 			# Translators: The status shown for an addon that has been marked as removed, before NVDA has been restarted.
 			statusList.append(_("Removed after restart"))
-		elif addon.isPendingDisable or (not addon.isPendingEnable and addon.isPendingInstall and addon.isDisabled):
+		elif addon.isPendingDisable or (addon.isDisabled and not addon.isPendingEnable
+			and (addon.isPendingInstall or globalVars.appArgs.disableAddons)):
 			# Translators: The status shown for an addon when it requires a restart to become disabled
 			statusList.append(_("Disabled after restart"))
-		elif addon.isPendingEnable or (addon.isPendingInstall and not addon.isDisabled):
+		elif addon.isPendingEnable or (not addon.isDisabled
+			and (addon.isPendingInstall or globalVars.appArgs.disableAddons)):

```

Comment:
Why this change?

Reply:
Exact same reason as above, I'm not sure if I'm clear because it sounds tricky to explain, but we want to communicate that all add-ons are disabled during this run, but we also want to display if they will be enabled or disabled when you will restart NVDA without the `--disable-addons` flag.